handlers/tokeninfo/errors.go

```diff
@@ -6,22 +6,25 @@ import (
 	"net/http"
 )
 
-type TokenInfoError struct {
+// Error type is used to wrap standard error messages that can be easily marshaled to JSON
+type Error struct {
 	Error            string `json:"error"`
 	ErrorDescription string `json:"error_description"`
 	statusCode       int
 }
 
 var (
-	ErrInvalidRequest = TokenInfoError{"invalid_request", "Access Token not valid", http.StatusBadRequest}
-	ErrInvalidToken   = TokenInfoError{"invalid_token", "Access Token not valid", http.StatusUnauthorized}
+	// ErrInvalidRequest should be used whenever the receiver failed to parse the request
+	ErrInvalidRequest = Error{"invalid_request", "Access Token not valid", http.StatusBadRequest}
+	// ErrInvalidToken should be used whenever the receiver failed to validate a JWT Token
+	ErrInvalidToken = Error{"invalid_token", "Access Token not valid", http.StatusUnauthorized}
 )
 
-// error response format for the token info endpoint
-func Error(w http.ResponseWriter, terr TokenInfoError) {
+// WriteError will write the Error e to the response writer, marshaled as JSON, and with the respective Status Code
+func WriteError(w http.ResponseWriter, e Error) {
```

Comment:
i would try to accept the generic error type, too, and fallback to some default status code and the err.Error() as the description. Or if really only the tokeninfo.Error is acceptable, then the Write method could have it as a receiver, like err.Write(w http.ResponseWriter). just nitpicking, though.